advanced-cache.php

```diff
@@ -432,7 +512,7 @@ function add_debug_html_to_output( $debug_html ) {
 
 
 // Did we find a batcached page that hasn't expired?
-if ( isset($batcache->cache['time']) && ! $batcache->genlock && time() < $batcache->cache['time'] + $batcache->cache['max_age'] ) {
+if ( isset($batcache->cache['time']) && ! $batcache->genlock && ( time() < $batcache->cache['time'] + $batcache->cache['max_age'] || ( $batcache->use_stale && time() < $batcache->cache['time'] + $batcache->cache['max_age'] + $batcache->stale_max_age ) || ( $batcache->use_stale && $batcache->stale_max_age === 0 ) ) ) {
```

Comment:
I think there's a bug where if use_stale is set, but `fastcgi_finish_request()` doesn't exist, this patch will always respond with cached content, and the cache will never get cleared/updated.